python/paddle/distributed/passes/pass_base.py

```diff
@@ -164,9 +166,61 @@ def _fusion_opt_last_rule(pass_before, pass_after):
         return True
 
 
+def _make_rule_from_white_lists_dict(before_white_lists_dict,
+                                     after_white_lists_dict):
+    def collect_pass_names(white_lists_dict, result):
+        for k, v in white_lists_dict.items():
+            result.add(k)
+            assert isinstance(v, (list, tuple))
+            for pass_name in v:
+                assert isinstance(pass_name, (bytes, str))
+                result.add(pass_name)
+
+    all_pass_names = set()
+    collect_pass_names(before_white_lists_dict, all_pass_names)
+    collect_pass_names(after_white_lists_dict, all_pass_names)
+
+    compatible_pass_dict = {}
+    for pass_name in all_pass_names:
+        compatible_pass_dict[pass_name] = set()
+
+    for k, v in before_white_lists_dict.items():
+        for pass_name in v:
+            compatible_pass_dict[k].add(pass_name)
+
+    for k, v in after_white_lists_dict.items():
+        for pass_name in v:
+            compatible_pass_dict[pass_name].add(k)
+
+    def rule(pass_before, pass_after):
+        all_passes_before = compatible_pass_dict.get(pass_after.name)
```

Comment:
What is the meaning of  `all_passes_before`?

Reply:
All passes that can be applied before `pass_after`.

Reply:
Thanks for pointing out that I use the wrong variable name here. I would fix this in the next PR.

Reply:
Done.